packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -383,12 +383,11 @@ export const fetchAndSaveMetadata =
         const provider = selectSelectedProviderForLabels(getState());
         if (!provider) return;
 
-        const selectedDeviceState = selectDevice(getState())?.state;
-        const deviceState = deviceStateArg || selectedDeviceState;
+        let device = deviceStateArg
```

Comment:
nitpick, I just try to avoid `let` when I can, so I would suggest to define some `getDevice(deviceStateArg?: string)` function and use it on both places to set device const (here and inside try after the await, in catch should be fine to keep the deviceState unrefreshed)

Reply:
this is again the conditional selector you suggested in https://github.com/trezor/trezor-suite/pull/9560#discussion_r1363952168 but I'd leave this for some next iteration..

Reply:
this is again the conditional selector you suggested in https://github.com/trezor/trezor-suite/pull/9560#discussion_r1363952168 but I'd leave this for some next iteration..

Reply:
yeah, I understand...